OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs

```diff
@@ -70,12 +70,19 @@ public WithInfantryBody(ActorInitializer init, WithInfantryBodyInfo info)
 
 			DefaultAnimation = new Animation(init.World, rs.GetImage(self), RenderSprites.MakeFacingFunc(self));
 			rs.Add(new AnimationWithOffset(DefaultAnimation, null, () => IsTraitDisabled));
+			PlayStandAnimation(self);
 
-			DefaultAnimation.PlayFetchIndex(NormalizeInfantrySequence(init.Self, info.StandSequences.Random(Game.CosmeticRandom)), () => 0);
 			state = AnimationState.Waiting;
 			move = init.Self.Trait<IMove>();
 		}
 
+		public void PlayStandAnimation(Actor self)
+		{
+			var sequence = DefaultAnimation.GetRandomExistingSequence(Info.StandSequences);
+			if (!string.IsNullOrEmpty(sequence))
```

Comment:
A null check is sufficient here.  The earlier HasSequences will filter out any bogus (i.e. empty) sequences.